cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/AbstractContext.java

```diff
@@ -230,6 +231,11 @@ public DynamicDatasetCache getDatasetCache() {
 
   protected <T extends Dataset> T getDataset(String namespace, String name, Map<String, String> arguments,
                                              AccessType accessType) throws DatasetInstantiationException {
+    if (NamespaceId.SYSTEM.getNamespace().equalsIgnoreCase(namespace)) {
+      throw new DatasetInstantiationException(String.format("Dataset %s cannot be instantiated from %s namespace. " +
```

Comment:
Any way to get the user id in this message? Should be able to inject ``AuthenticationContext``. 

``String.format("User %s cannot access dataset %s in %s namespace", authenticationContext.getPrincipal(), name, namespace)``?